core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -357,7 +357,8 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    */
   def setLogLevel(logLevel: String) {
     val validLevels = Seq("ALL", "DEBUG", "ERROR", "FATAL", "INFO", "OFF", "TRACE", "WARN")
-    if (!validLevels.contains(logLevel)) {
+    // let's allow lowcase or mixed case too
+    if (!validLevels.contains(logLevel.toUpperCase(Locale.ENGLISH))) {
```

Comment:
You could use `require` here even.
My suggestion was to use a `Set` above, and, to make it a static member of an object. However, maybe that's really trivial since this method is called so infrequently.
I do think it's best to pass the value to log4j that is validated here -- the uppercase version.